src/ice/mod.rs

```diff
@@ -443,6 +455,53 @@ mod test {
         }
     }
 
+    #[test]
+    pub fn candidate_pair_of_same_kind_does_not_get_nominated() {
+        let mut a1 = TestAgent::new(info_span!("L"));
+        let mut a2 = TestAgent::new(info_span!("R"));
+
+        let c1 = relay("1.1.1.1:1000", "udp");
+        a1.add_local_candidate(c1.clone());
+        a2.add_remote_candidate(c1);
+        let c2 = srflx("4.4.4.4:1000", "3.3.3.3:1000", "udp");
+        let c3 = host("3.3.3.3:1000", "udp");
```

Comment:
I didn't manage to reproduce this using `host` candidates alone so I re-built the scenario that we were seeing in production: relay candidates talking to srflx candidates and discovering new relay candidates.

Reply:
Nice to have a real world scenario here!